lib/logstash/filters/xml.rb

```diff
@@ -194,4 +204,26 @@ def filter(event)
     filter_matched(event) if matched
     @logger.debug? && @logger.debug("Event after xml filter", :event => event)
   end
+
+  private
+
+  def xml_parse_options
+    return Nokogiri::XML::ParseOptions::DEFAULT_XML unless @parse_options # (RECOVER | NONET)
+    @xml_parse_options ||= begin
```

Comment:
Only a question @kares, today in the docs we say that supported options are:
-  _strict_
- _no_warning_
- _no_error_
the ones provided by Nokojiri parser.
In the future if we update the parser version, and the parser changes/updates the list of flag  without we synch the documentation, then we have a misalignment. Do you think is best to have strict check of the flags here instead of relay on the validation done by the parser library, also if it means repeating it?

Reply:
here's the reasoning: initially I thought about only having `strict => true` option.
but than the options allow finer control, for whatever scenarios we might run into ... 
(e.g. might wan to allow network access for the parser to download schemas etc.)

so there's more parser options we can set, I purposefully only documented these few.
maybe even *no_error* and *no_warning* might be a bit too much as what is a warning/error
depends on the parser (might not  be clear).

being able to (unoficially) set all of the internal ones is the reason why I decided not to validate against a fixed set of options (that would be enumerated here).

going to remove *no_error* and *no_warning* from the docs and only document *strict* for now.
and checking that the official `parser_options` work (`'strict'`) will be covered by specs.

does that sound okay or do you have concerns?

Reply:
It's ok, go with the `strict` as `parser_options`. My thoughts was based on the assumption that everything we provide should be documented, if we let the door open so that someone can tinkering with undocumented behaviours, then in case we change library we potentially break more pipeline configurations than expected.